akka-actor/src/main/scala/akka/serialization/Serialization.scala

```diff
@@ -389,13 +389,38 @@ class Serialization(val system: ExtendedActorSystem) extends Extension {
 
     system.dynamicAccess.createInstanceFor[Serializer](fqn, List(classOf[ExtendedActorSystem] -> system)).recoverWith {
       case _: NoSuchMethodException =>
-        system.dynamicAccess.createInstanceFor[Serializer](fqn, Nil).recoverWith {
-          case e: NoSuchMethodException =>
-            if (bindingName == "") throw e // compatibility with (public) serializerOf method without bindingName
-            else
-              system.dynamicAccess.createInstanceFor[Serializer](
-                fqn,
-                List(classOf[ExtendedActorSystem] -> system, classOf[String] -> bindingName))
+        system.dynamicAccess.createInstanceFor[Serializer](fqn, List(classOf[ActorSystem] -> system)).recoverWith {
+          case _: NoSuchMethodException =>
+            system.dynamicAccess
+              .createInstanceFor[Serializer](fqn, List(classOf[ClassicActorSystemProvider] -> system))
+              .recoverWith {
+                case _: NoSuchMethodException =>
+                  system.dynamicAccess.createInstanceFor[Serializer](fqn, Nil).recoverWith {
+                    case e: NoSuchMethodException =>
+                      if (bindingName == "")
+                        throw e // compatibility with (public) serializerOf method without bindingName
+                      else
+                        system.dynamicAccess
+                          .createInstanceFor[Serializer](
+                            fqn,
+                            List(classOf[ExtendedActorSystem] -> system, classOf[String] -> bindingName))
+                          .recoverWith {
+                            case _: NoSuchMethodException =>
+                              system.dynamicAccess
+                                .createInstanceFor[Serializer](
+                                  fqn,
+                                  List(classOf[ActorSystem] -> system, classOf[String] -> bindingName))
+                                .recoverWith {
+                                  case _: NoSuchMethodException =>
+                                    system.dynamicAccess.createInstanceFor[Serializer](
+                                      fqn,
+                                      List(
+                                        classOf[ClassicActorSystemProvider] -> system,
+                                        classOf[String] -> bindingName))
```

Comment:
If we cascade down over all the possibilities and fail to find a constructor with one of the supported shapes, the user will get a `NoSuchMethodException` for `ClassicActorSystemProvider` and `String` giving the impression that this last one is the only one we support.

It would be better if we had a more personalized message. Like: none of the support constructors were found. And eventually a link to some docs?

Reply:
ok, added https://github.com/akka/akka/pull/31936/commits/0eae50659416ea903b75fbcc3c172d320c8c8323